javalib/src/main/scala/java/nio/Buffer.scala

```diff
@@ -1,6 +1,14 @@
 package java.nio
 
 abstract class Buffer private[nio] (val _capacity: Int) {
+  private[nio] type ElementType
+
+  private[nio] type BufferType >: this.type <: Buffer {
+    type ElementType = Buffer.this.ElementType
+  }
```

Comment:
Do we need this? Can't we replace the occurrences with singleton types (which will get erased to the underlying buffer type).